core/src/banking_stage/consumer.rs

```diff
@@ -1804,70 +1804,62 @@ mod tests {
-                    let deserialized_packets =
-                        unprocessed_packet_batches::transactions_to_deserialized_packets(
-                            &transactions,
-                        )
-                        .unwrap();
-                    assert_eq!(deserialized_packets.len(), num_conflicting_transactions);
-                    let mut buffered_packet_batches =
-                        UnprocessedTransactionStorage::new_transaction_storage(
-                            UnprocessedPacketBatches::from_iter(
-                                deserialized_packets.into_iter(),
-                                num_conflicting_transactions,
-                            ),
-                            ThreadType::Transactions,
-                        );
-                    consumer.consume_buffered_packets(
-                        &bank_start,
-                        &mut buffered_packet_batches,
-                        &BankingStageStats::default(),
-                        &mut LeaderSlotMetricsTracker::new(0),
-                    );
+            let consumer = Consumer::new(committer, recorder, QosService::new(1), None, None);
+
+            let num_conflicting_transactions = transactions.len();
+            let deserialized_packets =
+                unprocessed_packet_batches::transactions_to_deserialized_packets(&transactions)
+                    .unwrap();
+            assert_eq!(deserialized_packets.len(), num_conflicting_transactions);
+            let mut buffered_packet_batches =
+                UnprocessedTransactionStorage::new_transaction_storage(
+                    UnprocessedPacketBatches::from_iter(
+                        deserialized_packets.into_iter(),
+                        num_conflicting_transactions,
+                    ),
+                    ThreadType::Transactions,
+                );
+            let banking_stage_stats = BankingStageStats::default();
+            consumer.consume_buffered_packets(
+                &bank_start,
+                &mut buffered_packet_batches,
+                &banking_stage_stats,
+                &mut LeaderSlotMetricsTracker::new(0),
+            );
+
+            // After consume - check that all transactions were processed, and that the
+            // number of rebuffered transactions is correct - i.e. transactions were
+            // processed in batches of one.
+            assert!(buffered_packet_batches.is_empty());
+            assert_eq!(
+                banking_stage_stats
+                    .consumed_buffered_packets_count
+                    .load(Ordering::Relaxed),
+                num_conflicting_transactions
+            );
+            // Use bank to check the number of entries (batches)
+            assert_eq!(bank_start.working_bank.transactions_per_entry_max(), 1);
+            assert_eq!(
+                bank_start.working_bank.transaction_entries_count(),
+                num_conflicting_transactions as u64
+            );
```

Comment:
The `bank` tells us how many entries and the maximum size of entries. We can use these to verify that we processed them in the expected way.

Reply:
Looks like the test is changed from ensuring receive/process/submit one batch at a time (by checking test_fn is called between batches) to only assure number of batches landed in bank. 

I mean if consumer does "receive/process and cache batches, until receiveing buffer is empty. Then submit cached batches to bank at once". It won't pass original test, but will pass new one. Will it?

Maybe can question validity of this test. But its name `_interrupted` doesn't help in understanding the test's intention.

Reply:
Test is probably poorly named now. Originally (or at least [before MI](https://github.com/solana-labs/solana/commit/ee2f760d3d9040e3c4a642614279d89076001a34#diff-bb8464c2f02c8863fb60e6e29f023f3e6da93a93fbb9199a70a44c9c1b41c7f3L3888)) the test was testing behavior around conflicting txs on the same thread.

Before MI, the behavior was that a call to `consume_buffered_packets` only processed the first non-conflicting txs, in this case each iteration would process 1 packet. This is probably where the "interrupt" name comes from.

With the MI PR, that behavior obviously changed and we now processed all the packets in a single call to `consume_buffered_packets`.

> I mean if consumer does "receive/process and cache batches, until receiveing buffer is empty. Then submit cached batches to bank at once". It won't pass original test, but will pass new one. Will it?

It would still pass in that case as long as it called the `test_fn` once per batch, regardless of what it did internally. Similarly, with the changes I've made in this PR, the test will pass regardless of whether you cached batches or process + commit for each, as long as the result is the bank having all the entries of size 1.

Though, now that I'm re-looking at the history here. This test is really the same as `test_consume_buffered_packets` but with some additional checks...that are also always true for `test_consume_buffered_packets`. There's actually no need to have both anymore, but we should add these checks to `test_consume_buffered_packets`.

Reply:
>  It would still pass in that case as long as it called the test_fn once per batch ...

Oops, I wrote backwards in original comment - it would pass old test but not new one. 

But in any case, I agree w you that this test feels obsoleted since MI. Maybe combining it into other test as you suggested is a better idea.

Reply:
Yeah working on that right now. Going to delete this test, but also add another for testing retrys...since we actually lost that test case with MI woops!